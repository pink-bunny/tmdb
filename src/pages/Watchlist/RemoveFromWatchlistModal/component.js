import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Icon } from 'antd';

const RemoveFromWatchlistModalComponent = ({
  title,
  showModal,
  modalVisible,
  hideModal,
  removeFromWatchlist
}) => (
  <>
    <Icon type="delete" onClick={showModal} />

    <Modal
      visible={modalVisible}
      onCancel={hideModal}
      onOk={removeFromWatchlist}
      title="Delete list"
    >
      {`Do you want to delete ${title} from the watchlist?`}
    </Modal>
  </>
);

RemoveFromWatchlistModalComponent.propTypes = {
  title: PropTypes.string.isRequired,
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired,
  removeFromWatchlist: PropTypes.func.isRequired
};

export default RemoveFromWatchlistModalComponent;
