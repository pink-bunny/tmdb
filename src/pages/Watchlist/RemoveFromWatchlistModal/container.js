import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import withModal from '../../../hoc/withModal';
import { toggleToWatchlist as toggleToWatchlistAction } from '../../../../state/watchlist/actions';
import RemoveFromWatchlistModalComponent from './component';

class RemoveFromWatchlistModalContainer extends React.Component {
  removeFromWatchlist = () => {
    const {
      id,
      toggleToWatchlist
    } = this.props;
    const watchlist = false;
    const needRefetchWatchlist = true;

    toggleToWatchlist(id, watchlist, needRefetchWatchlist);
  }

  render() {
    const {
      modalVisible,
      hideModal,
      showModal,
      title
    } = this.props;

    return (
      <RemoveFromWatchlistModalComponent
        modalVisible={modalVisible}
        hideModal={hideModal}
        showModal={showModal}
        title={title}
        removeFromWatchlist={this.removeFromWatchlist}
      />
    );
  }
}

RemoveFromWatchlistModalContainer.propTypes = {
  toggleToWatchlist: PropTypes.func.isRequired,
  modalVisible: PropTypes.bool.isRequired,
  hideModal: PropTypes.func.isRequired,
  showModal: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired
};

const mapDispatchToPtops = {
  toggleToWatchlist: toggleToWatchlistAction
};

export default connect(null, mapDispatchToPtops)(withModal(RemoveFromWatchlistModalContainer));
