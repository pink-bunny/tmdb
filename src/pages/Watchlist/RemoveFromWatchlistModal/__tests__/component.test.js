import React from 'react';
import { shallow } from 'enzyme';

import RemoveFromWatchlistModal from '../component';

describe('RemoveFromWatchlistModal component', () => {
  const requiredProps = {
    title: 'Mock title',
    showModal: jest.fn(),
    modalVisible: true,
    hideModal: jest.fn(),
    removeFromWatchlist: jest.fn()
  };
  const wrapper = shallow(<RemoveFromWatchlistModal {...requiredProps} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */

  it('matches snapshot with required props', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
