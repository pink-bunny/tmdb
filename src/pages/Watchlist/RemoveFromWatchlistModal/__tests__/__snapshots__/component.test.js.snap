// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveFromWatchlistModal component matches snapshot with required props 1`] = `
<Fragment>
  <Icon
    onClick={[MockFunction]}
    type="delete"
  />
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[MockFunction]}
    onOk={[MockFunction]}
    title="Delete list"
    transitionName="zoom"
    visible={true}
    width={520}
  >
    Do you want to delete Mock title from the watchlist?
  </Modal>
</Fragment>
`;
